src/nvim/autocmd.c

```diff
@@ -2024,7 +2025,13 @@ char_u *getnextac(int c, void *cookie, int indent, bool do_concat)
   if (ac->exec.type == CALLABLE_CB) {
     typval_T argsin = TV_INITIAL_VALUE;
     typval_T rettv = TV_INITIAL_VALUE;
-    callback_call(&ac->exec.callable.cb, 0, &argsin, &rettv);
+    if (callback_call(&ac->exec.callable.cb, 0, &argsin, &rettv)) {
+      if (ac->exec.callable.cb.type == kCallbackLua) {
+        // If a Lua callback returns 'true' then the autocommand is removed
```

Comment:
So far the convention (at least with the decor provider) is to return false to cancel further callbacks.

Reply:
`nvim_buf_detach` uses `true` to detach a callback. Where do we use `false`?

Reply:
Ah, sorry. Glossed over the PR desc. `on_win` decor provider callbacks use false. Forgot about `nvim_buf_attach`, so we don't have a consistent convention.

Reply:
Personally `false` feels better to me as it feels more like a cancellation, but that's probably just because I've used `on_win` a lot more.

Reply:
Though FWIW we can go either way.

Reply:
I prefer to think about this as "only count this command as executed _for the purpose of `once`_ if it returns something truthy" (i.e., return `false` to pretend it never happened).

Reply:
Although `nil` is falsy, so that may have unintended consequences for callbacks that don't return anything. I feel like forcing callbacks to explicitly return if they want to be removed adds noticeable overhead -- not sure if that is worth it?

(Alternative would be to add another option `once-but-only-if-it-worked` (working title), but that's not great, either.)

Reply:
> Although nil is falsy, so that may have unintended consequences for callbacks that don't return anything.

Not returning anything (or returning `nil`) is not the same as returning `false`. The C API (and Lua) can distinguish these just as we do for `on_win`.

Reply:
I agree it should work like existing buf_attach and provider callbacks. i e explicitly `return false` will remove it. but `return nil` (same as no return statement) will not.

Reply:
But `true` or `false`?